File: core/Sources/WeatherCore/Data/HourlyForecast.swift
//
// HourlyForecast.swift
//

import Foundation

public struct HourlyForecast: Codable, Hashable {
    public let date: Date
    public let temp: Float
    public let state: WeatherState

    public init(date: Date, temp: Float, state: WeatherState) {
        self.date = date
        self.temp = temp
        self.state = state
    }
}

File: android/app/src/main/java/com/readdle/weather/core/HourlyForecast.kt
package com.readdle.weather.core

import android.os.Parcelable
import com.readdle.codegen.anotation.SwiftValue
import kotlinx.parcelize.Parcelize
import java.util.Date

@SwiftValue @Parcelize
data class HourlyForecast(
    var date: Date = Date(),
    var temp: Float = 0.0f,
    var state: WeatherState = WeatherState.NONE
): Parcelable
